src/nle.c

```diff
@@ -46,32 +46,35 @@ vt_char_color_extract(TMTCHAR *c)
     signed char color = 0;
     switch (c->a.fg) {
     case (TMT_COLOR_DEFAULT):
-        color =
-            (c->c == 32) ? CLR_BLACK : CLR_GRAY; // ' ' is BLACK else WHITE
+        color = (c->c == ' ') ? CLR_BLACK : CLR_GRAY; /* ' ' is 0 else 7 */
         break;
     case (TMT_COLOR_BLACK):
-        color = (c->a.bold) ? NO_COLOR : CLR_BLACK; // c = 8:0
+        /* "bright black" is dark gray. Non-bright black is black and
+           therefore invisible on the assumed black background. NetHack
+           shouldn't use it. So we assume bright black, which is 8 (also,
+           confusingly, NO_COLOR, but that's an aside). */
+        color = 8;
```

Comment:
Does NH never use `CLR_BLACK`? In `colors.h` it is defined and in `mapglyph.c` it looks like its used for some kinds of explosions... Why do we need to be rid of this logic?

Reply:
Some objects and monsters are black. E.g. Orcish arrow (https://github.com/facebookresearch/nle/blob/master/src/objects.c#L117) or (duh!), black dragons and unicorns (https://github.com/facebookresearch/nle/blob/master/src/monst.c#L1219, https://github.com/facebookresearch/nle/blob/master/src/monst.c#L839).

Reply:
So yes, it very much uses `CLR_BLACK`. Before this change, it would display them as gray/default in the tty output (which is more or less a bug in NetHack, triggered by our non-standard macro definitions). After fixing this, `CLR_BLACK` might come out as `8` (= bright black) in the tty, again not the same as `CLR_BLACK` (which is 0). Hence the change in winrl.cc.

Reply:
OK, revised the changes to nle.c (it's now using the `BRIGHT` logic).